runtime/doc/api.txt

```diff
@@ -1336,7 +1336,12 @@ nvim_set_client_info({name}, {version}, {type}, {methods}, {attributes})
       • {type}        Must be one of the following values. Client libraries
                       should default to "remote" unless overridden by the
                       user.
-                      • "remote" remote client connected to Nvim.
+                      • "remote" remote client connected to Nvim via Neovim
+                        modified MessagePack-RPC that is not fully
+                        MessagePack-RPC compliant (out-of-ordered responses
+                        are not supported).
+                      • "msgpack-rpc" remote client connected to Nvim via
+                        fully MessagePack-RPC compliant protocol.
```

Comment:
This seems like an ok compromise for now. In the future we can revisit `nvim_chan_set` if needed.